File: HackerRank/Algorithms/3. Strings/Funny_string.cpp
#include <cmath>
#include <iostream>
using namespace std;

string check_for_funny_string(string s) {
    int i = 0;
    int j = s.size() - 1;
    while (i <= j) {
        if (abs(s[i] - s[i+1]) !=
            abs(s[j] - s[j-1])) {
            return "Not Funny";
        }
        ++i;
        --j;
    }
    return "Funny";
}

int main() {
    /* Enter your code here.
       Read input from STDIN.
       Print output to STDOUT */
    int test_case;
    ios::sync_with_stdio(false);
    cin >> test_case;
    string input;
    while (test_case--) {
        cin >> input;
        cout << check_for_funny_string(input) << endl;
    }
    return 0;
}

File: HackerRank/Algorithms/3. Strings/Funny_string.java
import java.util.Scanner;
import java.lang.Math;

public class Solution {

    public static boolean checkForFunnyString(String s) {
        int i = 0;
        int j = s.length() - 1;
        while (i <= j) {
            if (Math.abs(s.charAt(i) - s.charAt(i+1)) !=
                Math.abs(s.charAt(j) - s.charAt(j-1))) {
                return false;
            }
            ++i;
            --j;
        }
        return true;
    }

    public static void main(String[] args) {
        /* Enter your code here.
           Read input from STDIN.
           Print output to STDOUT.
           Your class should be named Solution. */
        Scanner in = new Scanner(System.in);
        int testCase;
        String input;
        testCase = in.nextInt();
        in.nextLine();
        while (testCase != 0) {
            input = in.nextLine();
            if (checkForFunnyString(input))
                System.out.println("Funny");
            else
                System.out.println("Not Funny");
            --testCase;
        }
        in.close();
    }
}
